### PR TITLE
Put intermediate docker images of release pipelines in a dedicated ECR bucket

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Put intermediate docker images of release pipelines in a dedicated ECR bucket.
Typically, agent docker images are stored in `486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent/agent`.
For release pipelines, the docker images should be stored in `486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent/agent-release` instead (`-release` suffix added).

### Motivation

Agent docker images are stored in an ECR bucket in which they are kept only a limited amount of time.
It happens that we need them few days after they have been generated:
* because the release pipeline was launched one day, but we wanted to wait few days befre publishing the final artifacts.
* because we need to publish a new internal image from a release which was out few weeks ago.
In those situations, it happens that the ECR CI internal images have already been cleaned up, which forces up to generate them again, which takes time and raises some reproducibility concerns.

### Additional Notes

This change requires DataDog/cloudops#25155.

### Describe how to test your changes

Check in GitLab logs that internal images for release pipelines are put in `agent-release` instead of `agent` on ECR.
And check on ECR that only the release pipelines’ images are stored there. (So that evictions happen after a longer period of time)

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Comment by sgnn7** [comment]
Moving to 7.32.0 milestone since the freeze is today.

**Comment by L3n41c** [comment]
[Here is a successful PR pipeline with this change](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/5514149).

File: .gitlab/image_deploy/docker_linux.yml
Comment (albertvaka) [inline .gitlab/image_deploy/docker_linux.yml:190]: I see you are replacing all uses of the `trigger` gitlab keyword with regular jobs that call `inv pipeline.trigger-child-pipeline` (as defined in `docker_publish_job_definition`).

This will still work, but it has the following implications:
- The child pipelines might take longer to get triggered if all workers are busy (since these jobs now need a runner to trigger the child pipeline from bash)
- If the child pipeline fails, the job on the main pipeline will still appear green (because it succeeded triggering the child pipeline, even if the child pipeline failed). This is something we could/should fix inside `trigger-child-pipeline`, by waiting for the child pipeline to complete and bubbling up the status, but until we do that it might hide errors.

**Review by albertvaka — APPROVED** [review]
LGTM once the cloudops PR has been merged. Apart from some minor nits, I left a note about the implications of using `trigger-child-pipeline` instead of the `trigger` gitlab keyword. I'm not sure about the reason behind that change, but if we go that way we might want to change a bit `trigger-child-pipeline` so it watches the child pipelines it triggers.

**Review by vboulineau — APPROVED** [review]

File: .gitlab/image_build/docker_windows.yml
Comment (L3n41c) [reply .gitlab/image_build/docker_windows.yml]: Good catch !
I’ve removed this default value.

File: .gitlab/docker_common/publish_job_templates.yml
Comment (L3n41c) [reply .gitlab/docker_common/publish_job_templates.yml]: `size:large` removed !

File: .gitlab/image_deploy/docker_linux.yml
Comment (L3n41c) [reply .gitlab/image_deploy/docker_linux.yml:190]: The initial motivation for using `inv pipeline.trigger-child-pipeline` instead of the `trigger` GitLab keyword was that the `trigger` keyword doesn’t allow complex logic for setting environment variables.
With this change, pretty much all jobs need the logic of suffixing the docker bucket with `-release` if the build is a build of a tagged commit.
So, this logic has been factorised in [`.docker_publish_job_definition`](https://github.com/DataDog/datadog-agent/pull/8560/files#diff-239bf1835b577f11028b131aadce4b3b609380c7ebcc9f01d3ddef3dcac93212R14).

**Comment by L3n41c** [comment]
[Here is a successful PR pipeline with this change](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/5611629).

**Comment by albertvaka** [comment]
Thanks a lot for implementing this 🎉

**Pull request merged**
